modules/data.atmosphere/R/met_temporal_downscale.Gaussian_ensemble.R

```diff
@@ -0,0 +1,398 @@
+      lowday <- (x - wd) * div
+      highday <- (x + wd) * div
+      if (lowday < 0) {
+        lowday <- 0
+      }
+      if (highday > reso_len) {
+        highday <- reso_len
+      }
+      four <- vector()
+      four <- rand_vect_cont(div, source$precipitation_flux[x], sd = sd(train$precipitation_flux[lowday:highday]))
+      four[four < 0] <- 0
+      precip <- append(precip, four)
+    }
+    df$precipitation_flux <- precip
+    
+    # Specific Humidity, eastward wind and northward wind
+    wnd <- c("specific_humidity", "eastward_wind", "northward_wind", "surface_downwelling_longwave_flux_in_air", 
+             "air_pressure")
+    for (u in wnd) {
+      woo <- vector()
+      a <- as.numeric(train[[u]])
+      sour <- as.numeric(source[[u]])
+      if (all(is.na(sour)) == TRUE) {
+        woo <- rep(NA, reso_len)
+      }
+      if (all(is.na(sour)) == FALSE) {
+        for (x in seq_along(sour)) {
+          lowday <- (x - wd) * div
+          highday <- (x + wd) * div
+          if (lowday < 0) {
+            lowday <- 0
+          }
+          if (highday > reso_len) {
+            highday <- reso_len
+          }
+          four <- vector()
+          for (n in seq_along(1:div)) {
+            four[n] <- rnorm(1, mean = sour[x], sd = sd(a[lowday:highday]))
+          }
+          woo <- append(woo, four)
+        }
+      }
+      df[1:length(woo), u] <- woo
+    }
+    
+    df$specific_humidity[df$specific_humidity < 0] <- 0
+    
+    # Downwelling shortwave radiation flux Ephemeris is a function to calculate
+    # sunrise/sunset times and daylength for SW calculations in sine swdn_method
+    ephemeris <- function(lat, lon, date, span = 1, tz = "UTC") {
+      
+      lon.lat <- matrix(c(lon, lat), nrow = 1)
+      
+      # using noon gets us around daylight saving time issues
+      day <- as.POSIXct(sprintf("%s 12:00:00", date), tz = tz)
+      sequence <- seq(from = day, length.out = span, by = "days")
+      
+      sunrise <- maptools::sunriset(lon.lat, sequence, direction = "sunrise", 
+                                    POSIXct.out = TRUE)
+      sunset <- maptools::sunriset(lon.lat, sequence, direction = "sunset", 
+                                   POSIXct.out = TRUE)
+      solar_noon <- maptools::solarnoon(lon.lat, sequence, POSIXct.out = TRUE)
+      
+      data.frame(date = as.Date(sunrise$time), sunrise = as.numeric(format(sunrise$time, "%H%M")), 
+                 solarnoon = as.numeric(format(solar_noon$time, "%H%M")), 
+                 sunset = as.numeric(format(sunset$time, "%H%M")),
+                 day_length = as.numeric(sunset$time - sunrise$time))
+    }
+    
+    swsource <- source$surface_downwelling_shortwave_flux_in_air
+    swdn <- vector()
+    
+    # The sine swdn_method produces an hourly sine wave of
+    if (swdn_method == "sine") {
+      
+      eph <- ephemeris(lat_train, lon_train, date = paste0(eph_year, "-01-01 00:00:00"), 
```

Comment:
Yeah, really hadn't considered that. I'll change eph_year to be the year - 1 if it's a leap year